cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -80,6 +81,18 @@ public void onCallback(NotificationContext notificationContext) {
     }
   }
 
+  @Override
+  public void onConfigChange(List<InstanceConfig> configs, NotificationContext changeContext) {
+    if (updateDisabledHosts()) {
+      generateShardConfig();
+    }
+  }
+
+  @Override
+  public void onExternalViewChange(List<ExternalView> externalViewList, NotificationContext changeContext) {
```

Comment:
ideally we should use the externalViewList passed in to the callback to reduce the load on zk.